src/yamlfix/adapters.py

```diff
@@ -693,3 +705,33 @@ def _restore_jinja_variables(source_code: str) -> str:
             fixed_source_lines.append(line)
 
         return "\n".join(fixed_source_lines)
+
+    def _fix_comment_only_files(self, source_code: str) -> str:
+        """Add a mapping key with an id to the start of the document\
+            to preserve comments."""
+        fixed_source_lines = []
+        yamlfix_document_id_line = f"yamlfix_{self.config.document_fix_id}:"
+
+        # Add the document id line after each document start
+        has_start_indicator = False
+        for line in source_code.splitlines():
+            fixed_source_lines.append(line)
+            if line.startswith("---"):
+                has_start_indicator = True
+                fixed_source_lines.append(yamlfix_document_id_line)
```

Comment:
For performance reasons it would be nice if we could stop the iteration on the file source as soon as we've done the required changes. Same applies at the `_restore_comment_only_files`.

Maybe reformulate it so that once the condition is met it injects the required line and exits. 

Now that I see it, although this is a good idea, it may need a non-trivial refactor where the fixers tweak directly the parts of the code from `self.` instead of receiving the source as an argument and returning as a return value.

If you don't feel like doing this now we can open an issue to track it for the future